src/hotspot/cpu/riscv/interp_masm_riscv_64.cpp

```diff
@@ -541,31 +556,24 @@ void InterpreterMacroAssembler::index_check_without_pop(Register Rarray, Registe
     //null_check_throw(Rarray, arrayOopDesc::length_offset_in_bytes(), /*temp*/RsxtIndex); FIXME_RISCV
   }
 
-  // Rindex might contain garbage in upper bits (remember that we don't sign extend
-  // during integer arithmetic operations). So kill them and put value into same register
-  // where ArrayIndexOutOfBounds would expect the index in.
-  slli(RsxtIndex, Rindex, 32);
```

Comment:
why you have removed these operations?

Reply:
registers are always correctly sign-extended on riscv